src/runtime/server/plugins/refresh-token.server.ts

```diff
@@ -0,0 +1,39 @@
+import { _fetch } from '../../utils/fetch'
+import { jsonPointerGet } from '../../helpers'
+import { defineNuxtPlugin, useAuthState } from '#imports'
+export default defineNuxtPlugin({
+  name: 'refresh-token-plugin',
+  enforce: 'pre',
+  async setup (nuxtApp) {
+    // TODO: get rid of duplicate code
```

Comment:
I think you can resolve this now? 

You could add helpers into the `refresh/` composable folder and import them here - I guess it's mostly about pointer extraction?

If you want to go a bit further you could add a more generic pointer extraction helper that can be used here, in your `useAuth` composable and in the `local` `useAuth` composable